Introduction to xemc3.ipynb

```diff
@@ -0,0 +1,3171 @@
+       "          3.71417615, 3.66525302],\n",
+       "         ...,\n",
+       "         [5.59027424, 5.58100907, 5.56053413, ..., 3.7435559 ,\n",
+       "          3.69303175, 3.64369176],\n",
+       "         [5.59192682, 5.58422348, 5.56529866, ..., 3.74788951,\n",
+       "          3.69714638, 3.64790418],\n",
+       "         [5.59337562, 5.58724114, 5.56988255, ..., 3.75223099,\n",
+       "          3.701277  , 3.65213093]]])\n",
+       " Dimensions without coordinates: r, theta, phi,\n",
+       " <xarray.DataArray 'Te' (r: 139, theta: 512, phi: 36)>\n",
+       " [2562048 values with dtype=float64]\n",
+       " Dimensions without coordinates: r, theta, phi\n",
+       " Attributes:\n",
+       "     long_name:  Electron temperature\n",
+       "     units:      eV)"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "R = ds.R_bounds.mean(dim=(\"delta_r\", \"delta_theta\", \"delta_phi\"))\n",
+    "z = ds.z_bounds.mean(dim=(\"delta_r\", \"delta_theta\", \"delta_phi\"))\n",
+    "phi = ds.phi_bounds.mean(dim=\"delta_phi\")\n",
+    "x = R * np.cos(phi)\n",
+    "y = R * np.sin(phi)\n",
+    "x, ds.Te"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 125,
+   "id": "b2b57e7d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Total number of gridpoints in the W7X gridstructure is  2562048\n"
+     ]
+    }
+   ],
+   "source": [
+    "x.shape\n",
+    "tot_grid_points = x.shape[0]*x.shape[1]*x.shape[2]\n",
+    "print(\"Total number of gridpoints in the W7X gridstructure is \", tot_grid_points)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "13ee82c1",
+   "metadata": {},
+   "source": [
+    "## Use of NaN values in the parameter mesh\n",
+    "\n",
+    "Not all gridcells have a defined parameter value attached to it. This is mostly the outer and inner region of the machine where the values of many parameter has been left out because this is not the regions where the interesting physics happen.\n",
+    "This is illustrated in the above plot example of the electron temperature $T_e$. In the cell below you can se how large a fraction of the total number of gridpoints the parameter mesh for the electron temperature that has NaN as a value."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 130,
+   "id": "0b63deae",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "How many nans in the Te mesh?  (584972,)\n",
+      "Fraction of nans with respect to gridpoints  0.22832202987609912\n"
+     ]
+    }
+   ],
+   "source": [
+    "n_nans_Te = int(list(np.where(np.isnan(np.asarray(ds.Te)) == True)[0].shape)[0])\n",
+    "print(\"How many nans in the Te mesh? \", np.where(np.isnan(np.asarray(ds.Te)) == True)[0].shape)\n",
```

Comment:
There is no "Te mesh" - there is just one mesh.

Also, the mesh does not contain nan's - just the `Te` field contains nans.